build.rs

```diff
@@ -14,6 +15,141 @@ use walkdir::WalkDir;
 
 const MLN_GIT_REPO: &str = "https://github.com/maplibre/maplibre-native.git";
 const MLN_REVISION: &str = "8e3899df391b21f9cd1786194cb6fbfa318795da";
+const MLN_RELEASE_TAG: &str = "core-12e0922fc4cadcd88808830e697cfb1d5206c8c9";
+
+#[derive(Debug, Clone, serde::Deserialize)]
+struct ReleaseAsset {
+    name: String,
+    browser_download_url: String,
+    digest: Option<String>,
+}
+
+impl ReleaseAsset {
+    fn hash(&self) -> Option<&str> {
+        self.digest.as_ref()?.strip_prefix("sha256:")
+    }
+}
+
+#[derive(serde::Deserialize)]
+struct Release {
+    assets: Vec<ReleaseAsset>,
+}
+
+/// get release metadata and search for wanted files
+fn fetch_asset_metadata(
+    release_tag: &str,
+    target_filenames: &[&str],
+) -> Result<Vec<ReleaseAsset>, Box<dyn std::error::Error>> {
+    let api_url = format!(
+        "https://api.github.com/repos/maplibre/maplibre-native/releases/tags/{release_tag}"
+    );
+
+    let response = ureq::get(&api_url).call()?;
+    let body = response.into_string()?;
+    let release: Release = serde_json::from_str(&body)?;
+    let assets = release.assets;
+
+    let mut found_assets = Vec::new();
+    let mut missing_files = Vec::new();
+
+    for filename in target_filenames {
+        if let Some(asset) = assets.iter().find(|a| a.name == *filename).cloned() {
+            found_assets.push(asset);
+        } else {
+            missing_files.push(*filename);
+        }
+    }
+
+    if !missing_files.is_empty() {
+        return Err(format!(
+            "Files not found in release {}: {}",
+            release_tag,
+            missing_files.join(", ")
+        )
+        .into());
+    }
+
+    Ok(found_assets)
+}
+
+/// determine if asset needed
```

Comment:
could you go over these doc comments again, If properly filled out, these can be very usefull for telling the next person what something does.